Sources/Publish/API/AnyDateFormatter.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AnyDateFormatter.swift
+//  
+//
+//  Created by Dustin Pfannenstiel on 3/10/20.
+//  Copyright (c) Dustin Pfannenstiel, 2020
+//  MIT license, see LICENSE file for details
+//
+
+import Foundation
+
+/// A common interface to abstract common methods and properties present
+/// in all date formatter types.
+public protocol AnyDateFormatter: Formatter {
+    /// The time zone for the receiver.
+    var timeZone: TimeZone! { get set }
+    /// The date format string used by the receiver.
+    var dateFormat: String! { get }
+
+    /// Returns a date representation of a given string interpreted using the receiver’s current settings.
+    /// - parameter string: The string to parse.
+    func date(from string: String) -> Date?
+    /// Returns a string representation of a given date formatted using the receiver’s current settings.
+    /// - parameter date: The date to format.
+    func string(from date: Date) -> String
```

Comment:
This API is not used within Publish, so including it feels a bit like premature generalization, and makes the `AnyDateFormatter` protocol more complex than it has to be.